src/toolkits/object_detection/one_shot_object_detection/one_shot_object_detector.cpp

```diff
@@ -63,181 +62,208 @@ flex_dict build_annotation( ParameterSampler &parameter_sampler,
+  float bounding_box_width = max_x - min_x;
   float bounding_box_height = max_y - min_y;
 
   flex_dict coordinates = {std::make_pair("x", center_x),
                            std::make_pair("y", center_y),
                            std::make_pair("width", bounding_box_width),
-                           std::make_pair("height", bounding_box_height)
-                          };
+                           std::make_pair("height", bounding_box_height)};
   flex_dict annotation = {std::make_pair("coordinates", coordinates),
-                          std::make_pair("label", label)
-                         };
+                          std::make_pair("label", label)};
   return annotation;
 }
 
-static std::map<std::string,size_t> generate_column_index_map(
-    const std::vector<std::string>& column_names) {
-    std::map<std::string,size_t> index_map;
-    for (size_t k=0; k < column_names.size(); ++k) {
-        index_map[column_names[k]] = k;
-    }
-    return index_map;
+static std::map<std::string, size_t> generate_column_index_map(
+    const std::vector<std::string> &column_names) {
+  std::map<std::string, size_t> index_map;
+  for (size_t k = 0; k < column_names.size(); ++k) {
+    index_map[column_names[k]] = k;
+  }
+  return index_map;
 }
 
-boost::gil::rgba8_image_t::view_t create_starter_image_view(flex_image &object_input) {
+boost::gil::rgba8_image_t::view_t create_starter_image_view(
+    const flex_image &object_input) {
   DASSERT_TRUE(object_input.is_decoded());
-  flex_image object = image_util::resize_image(object_input,
-                                               object_input.m_width,
-                                               object_input.m_height,
-                                               4,
-                                               true).to<flex_image>();
+  flex_image object =
+      image_util::resize_image(object_input, object_input.m_width,
+                               object_input.m_height, 4, true)
+          .to<flex_image>();
   DASSERT_TRUE(object.is_decoded());
   DASSERT_EQ(object.m_channels, 4);
   boost::gil::rgba8_image_t::view_t starter_image_view = interleaved_view(
-    object.m_width,
-    object.m_height,
-    (boost::gil::rgba8_pixel_t*) (object.get_image_data()),
-    object.m_channels * object.m_width // row length in bytes
+      object.m_width, object.m_height,
+      (boost::gil::rgba8_pixel_t *)(object.get_image_data()),
+      object.m_channels * object.m_width  // row length in bytes
   );
   return starter_image_view;
 }
 
+std::pair<flex_image, flex_dict>
+create_synthetic_image_from_background_and_starter(const flex_image &starter,
+                                                   const flex_image &background,
+                                                   std::string label,
+                                                   size_t seed,
+                                                   size_t row_number) {
+  // if (!(starter.is_decoded())) {
```

Comment:
Do we still need this?

Reply:
Nope, removed, thanks for pointing it out!